plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java

```diff
@@ -79,31 +80,32 @@ protected void channelRead0(ChannelHandlerContext ctx, Message msg) throws Excep
                 log.error("Fail to encrypt message: ", e.getMessage());
             }
         }
-
-        boolean sawKtTag = false;
+        String kafkaTopic = defaultTopic;
+        // Producer will simply do round-robin when a null partitionKey is provided
+        byte[] partitionKey = null;
 
         for (String tag : msg.getTags()) {
             if (tag.startsWith("kt:")) {
-                sawKtTag = true;
-                sendOrReportFailure(tag.substring(3), payload);
+                kafkaTopic = tag.substring(3);
+            } else if (tag.startsWith("pk:")) {
+              // Base64 decode the partitionKey to get the raw bytes
+                partitionKey = Base64.getUrlDecoder().decode( tag.substring(3));
```

Comment:
Why do we need to Base64 decode the partition key?  It seems from the apache KafkaProducer examples they just use a `String` as the key.  I'm guessing if we use a String we can still just call `hashcode()` on it.

Reply:
Even using String as the key, it will eventually be encoded into bytes. But I agree, we should let kafka producer to do the serialization of String using UTF-8. Will change the key to `String`